libraries/AP_RangeFinder/AP_RangeFinder_Benewake_TFMiniPlus.cpp

```diff
@@ -149,20 +150,31 @@ void AP_RangeFinder_Benewake_TFMiniPlus::update()
     }
 }
 
-bool AP_RangeFinder_Benewake_TFMiniPlus::process_raw_measure(le16_t distance_raw, le16_t strength_raw,
+void AP_RangeFinder_Benewake_TFMiniPlus::process_raw_measure(le16_t distance_raw, le16_t strength_raw,
                                                              uint16_t &output_distance_cm)
 {
     uint16_t strength = le16toh(strength_raw);
+    const uint16_t MAX_DIST_CM = 1200;
+    const uint16_t MIN_DIST_CM = 10;
 
     output_distance_cm = le16toh(distance_raw);
 
-    if (strength < 100 || strength == 0xFFFF) {
-        return false;
+    if (strength < 100 || strength == 0xFFFF || output_distance_cm > MAX_DIST_CM) {
+        /*
+         * From manual: "when the signal strength is lower than 100 or equal to
+         * 65535, the detection is unreliable, TFmini Plus will set distance
+         * value to 0." - force it to the max distance so status is set to OutOfRangeHigh
+         * rather than NoData.
+         */
+        output_distance_cm = MAX_DIST_CM;
```

Comment:
MAX_DIST_CM is the published normal value.
Some devices may show the published values.

Reply:
Not sure I understand this comment. MAX_DIST_CM is actually the maximum distance that can be measure by this specific sensor, as published by the vendor.

Reply:
If the sensor fails to measure, it returns an abnormal value.
It's not right to set a maximum value of 12 meters at that outlier.

Reply:
Yes, but see what I said below. This is not what the other sensors are doing. See libraries/AP_RangeFinder/AP_RangeFinder_Benewake.cpp:

```c++
    if (count_out_of_range > 0) {
        // if only out of range readings return larger of
        // driver defined maximum range for the model and user defined max range + 1m
        reading_cm = MAX(model_dist_max_cm(), max_distance_cm() + BENEWAKE_OUT_OF_RANGE_ADD_CM);
        return true;
    }
```

We don't count out of range measurements here because in the I2C backend we always have 1 sample. 
The changes are logically separate: 1) make them have the same behavior; 2) make them common to share code; 3) adapt better to the corner cases


These changes here are only for (1).